### PR TITLE
Crash game

### PR DESCRIPTION
**Pull request by TinelValentin**
Any comment about refactoring the class CrashView.xaml.cs will be ignored

### CODE REVIEW

File: SevenSlots/SevenSlots/SevenSlots/View/App.xaml.cs [hunk 1]
```diff
@@ -13,6 +13,8 @@ public App()
             DependencyService.Register<IUserRepository, UserRepository>();
             DependencyService.Register<IUserService, UserService>();
             InitializeComponent();
+
```
Comment (BrassoiCatalin) [inline SevenSlots/SevenSlots/SevenSlots/View/App.xaml.cs:16]: Why so many white spaces?

**Review by BrassoiCatalin — CHANGES_REQUESTED** [review]

**Review by BrassoiCatalin — APPROVED** [review]

File: SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            </Grid.RowDefinitions>
+            <StackLayout Grid.Row="0">
+                <CollectionView  x:Name="Results" ItemsSource="{Binding LastResults}" 
+                                  Margin="0,10,0,10" HeightRequest="50" >
+                    <CollectionView.Background>
+                        <LinearGradientBrush StartPoint="0, 0" EndPoint="1, 2">
+                            <GradientStop Color="#e36f6f" Offset="0.0"/>
+                            <GradientStop Color="#a62523" Offset="0.3"/>
+                            <GradientStop Color="#69100f" Offset="0.5"/>
+
```
Comment (Sorin37) [inline SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml]: empty new line

File: SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                           WidthRequest="500"
+                           HeightRequest="300"
+                           Opacity="1"
+                           HorizontalOptions="Start" />
+                    </Frame>
+                    <Label Margin="180, -280, 0, 0" FontSize="35" FontAttributes="Bold" TextColor="White" Text="{Binding MultiplierValue, StringFormat='{0}x'}"></Label>
+                    <Image Source="graph.png" Margin="40,-60,0,0" WidthRequest="380"/>
+                    <Image Source="seven.jpg" Margin="-220,-80,0,0" WidthRequest="40" Rotation="40" x:Name="rocket"/>
+                    <StackLayout Spacing="90" Margin="10,-220,0,0">
+                        <Label Style="{StaticResource MultiplierLabel}" Text="{Binding MultiplierLabelValues[2], StringFormat='{0}.0x'}"></Label>
```
Comment (Sorin37) [inline SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml:61]: brake attributes on multiple lines

File: SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+                        <Label Style="{StaticResource MultiplierLabel}" Text="{Binding MultiplierLabelValues[1], StringFormat='{0}.0x'}"></Label>
+                        <Label Style="{StaticResource MultiplierLabel}" Text="{Binding MultiplierLabelValues[0], StringFormat='{0}.0x'}"></Label>
+                    </StackLayout>
+                    <StackLayout Spacing="130" Margin="50,-10,0,0" Orientation="Horizontal">
+                        <Label Style="{StaticResource MultiplierLabel}" Text="{Binding SecondsPassedValues[0], StringFormat='{0}s'}"></Label>
+                        <Label Style="{StaticResource MultiplierLabel}" Text="{Binding SecondsPassedValues[1], StringFormat='{0}s'}"></Label>
+                        <Label Style="{StaticResource MultiplierLabel}" Text="{Binding SecondsPassedValues[2], StringFormat='{0}s'}"></Label>
+                    </StackLayout>
+                </StackLayout>
+                <StackLayout x:Name="PlaneCrashedText" IsVisible="false" Margin="20,-200,0,0">
```
Comment (Sorin37) [inline SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml]: no space after comma

File: SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+                </Grid.RowDefinitions>
+                <Label Grid.Row="0" 
+                   Grid.Column="0" 
+                   Text="Total Bet:" 
+                   Style="{StaticResource BlackjackLabel}"/>
+                <Label Grid.Row="0" 
+                   Grid.Column="1" 
+                   Text="{Binding Player.TotalBet, StringFormat='{0:c1}', FallbackValue=$0.00}" 
+                   Style="{StaticResource BlackjackLabel}"/>
+
```
Comment (Sorin37) [inline SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml]: empty new line

File: SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+        decimal multiplierExponent = 0.01m;
+        private int _countdown = 10;
+        private Player _player;
+        private bool _canBet;
+        private bool canCashOut = false;
+        private int _ownedMoney;
+        private int _lastWin;
+        private ObservableCollection<decimal> lastResults = new ObservableCollection<decimal>()
+        {
+            1.26m,1.25m
```
Comment (Sorin37) [inline SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs]: no space after comma

File: SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+        };
+        private List<int> multiplierLabelValues = new List<int>()
+        {
+            1,2,4
```
Comment (Sorin37) [inline SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs]: no space after comma

File: SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+        };
+        private List<int> secondsPassedValues = new List<int>()
+        {
+            0,1,2
```
Comment (Sorin37) [inline SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs]: no space after comma

File: SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+                        var power = MultiplierLabelValues.Count - i - 1;
+                        var value = (int)MultiplierValue - (int)Math.Pow(2, power) + 1;
+                        MultiplierLabelValues[i] = value;
+                        OnPropertyChanged(nameof(MultiplierLabelValues));
+
+                        if (crash)
+                        {
+                            MultiplierLabelValues = new List<int>()
+                    {
+                            1,2,4
```
Comment (Sorin37) [inline SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs:180]: no space after comma

File: SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs [hunk 10]
```diff
@@ -0,0 +1,286 @@
...
+                for (int i = 0; i < SecondsPassedValues.Count; i++)
+                {
+                    SecondsPassedValues[i] += 1;
+                    OnPropertyChanged(nameof(SecondsPassedValues));
+
+                    if (crash)
+                    {
+                        SecondsPassedValues = new List<int>()
+                    {
+                            0,1,2
```
Comment (Sorin37) [inline SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs:204]: no space after comma

File: SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs [hunk 11]
```diff
@@ -0,0 +1,286 @@
...
+                return true;
+            });
+        }
+
+        public Task StartCrashAsync()
+        {
+            UpdateMultiplierLabels();
+            UpdateSecondsLabels();
+            bool stopThread = false;
+#pragma warning disable CS4014 // This call is not awaited because we want the animation to run while the multiplier increases
```
Comment (Sorin37) [inline SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs:218]: fix indentation

File: SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs [hunk 12]
```diff
@@ -0,0 +1,286 @@
...
+            Fly();
+#pragma warning restore CS4014
```
Comment (Sorin37) [inline SevenSlots/SevenSlots/SevenSlots/View/CrashView.xaml.cs:220]: fix indentation

**Review by BrassoiCatalin — APPROVED** [review]

**Review by Sorin37 — APPROVED** [review]

**Pull request merged**